src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -48,54 +51,75 @@ public EntityType getFlattenedEntityType(UUID entityTypeId, boolean doFinalRenam
+      renamedAliases.put(oldAlias, newAlias);
+    }
 
-    List<EntityTypeColumn> finalColumns = new ArrayList<>();
     Set<String> finalPermissions = new HashSet<>(originalEntityType.getRequiredPermissions());
+    Stream.Builder<Stream<EntityTypeColumn>> columns = Stream.<Stream<EntityTypeColumn>>builder()
+      .add(copyColumns(sourceFromParent, originalEntityType, renamedAliases));
+
     for (EntityTypeSource source : originalEntityType.getSources()) {
-      if (source.getType().equals(DB_TYPE)) {
-        Pair<EntityTypeSource, List<EntityTypeColumn>> updatePair = getConvertedSourceAndColumns(originalEntityType, source, null, false); // TODO: think about this, may not be able to hardcode false here
-        flattenedEntityType.addSourcesItem(updatePair.getLeft());
-        finalColumns.addAll(updatePair.getRight());
-      } else {
+      if (source.getType().equals("db")) {
+        flattenedEntityType.addSourcesItem(copySource(sourceFromParent, source, renamedAliases, true));
+      }
+      else {
         UUID sourceEntityTypeId = UUID.fromString(source.getId());
-        EntityType flattenedSourceDefinition = getFlattenedEntityType(sourceEntityTypeId, false);
+        // Recursively flatten the source and add it to the flattened entity type
+        EntityType flattenedSourceDefinition = getFlattenedEntityType(sourceEntityTypeId, source);
         finalPermissions.addAll(flattenedSourceDefinition.getRequiredPermissions());
-
-        // If an entity type source contains multiple db sources, then we need to keep the original alias in order to
-        // distinguish the different targets. Frequently, it will likely only have one db source. In this case we
-        // can use the outer alias only, in order to keep field names more concise
-        boolean keepOriginalAlias = countDbSources(flattenedSourceDefinition) > 1;
-
-        for (EntityTypeSource subSource : flattenedSourceDefinition.getSources()) {
-          String oldAlias = subSource.getAlias();
-          Pair<EntityTypeSource, List<EntityTypeColumn>> updatePair = getConvertedSourceAndColumns(flattenedSourceDefinition, subSource, source, keepOriginalAlias);
-          String newAlias = updatePair.getLeft().getAlias();
-          if (!oldAlias.equals(newAlias)) {
-            updateOtherSources(oldAlias, newAlias, flattenedSourceDefinition.getSources());
-          }
-          flattenedEntityType.addSourcesItem(updatePair.getLeft());
-          finalColumns.addAll(updatePair.getRight());
-        }
+        // Add a prefix to each column's name and idColumnName, then add em to the flattened entity typeg
+        columns.add(
+          flattenedSourceDefinition.getColumns()
+            .stream()
+            .map(col -> col
+              .name(aliasPrefix + source.getAlias() + '.' + col.getName())
+              .idColumnName(col.getIdColumnName() == null ? null : aliasPrefix + source.getAlias() + '.' + col.getIdColumnName())
```

Comment:
Here's the fix for MODFQMMGR-406 - add the prefix to `idColumnName` in the same way we do it for column names. And by doing both right next to each other, it helps ensure that we're handling both in the same way (which is important, since ID column name needs to refer to valid columns, so when column names get updated, ID column names need the same treatment)